packages/indexer-client/src/types/clientTypes.ts

```diff
@@ -899,3 +899,35 @@ export interface GetIndexerFastWithdrawalSignatureResponse {
   txBytes: Hex;
   signatures: Hex[];
 }
+
+export interface IndexerStakingV2PoolSnapshot {
+  timestamp: BigDecimal;
+  cumulativeStaked: BigDecimal;
+  cumulativeUnstaked: BigDecimal;
+  numberOfStakers: BigDecimal;
+}
+
+export interface GetIndexerStakingV2PoolSnapshotsParams {
+  // Currently accepts all integers, in seconds
+  granularity: number;
+  // Seconds
+  maxTimeInclusive?: number;
+  limit: number;
+}
+export interface GetIndexerStakingV2PoolSnapshotsResponse {
+  snapshots: IndexerStakingV2PoolSnapshot[];
+}
+
+export interface IndexerStakingV2Staker {
+  address: Address;
+  stakedAmount: BigDecimal;
+  poolShare: BigDecimal;
```

Comment:
Could we add a JSDoc to explain that this is a fraction?

Also, for smaller numbers, we prob don't need `BigDecimal`. However, this does make it more consistent with other fields, so i'll leave the decision up to you